Update (#68)
OpenVoiceOS/community-docs: docs/config.md

# OVOS Configuration
When you first start OVOS, there should not be any configuration needed to have a working device.

> **NOTE** To continue with the examples, you will need access to a shell on your device.  This can be achevied with SSH.  Connect to your device with the command `ssh ovos@<device_ip_address>` and enter the password `ovos`.

Reviewer: ```suggestion
**NOTE** To continue with the examples, you will need access to a shell on your device.  This can be achevied with SSH.  Connect to your device with the command `ssh ovos@<device_ip_address>` and enter the password `ovos`.
```
```suggestion
**NOTE** To continue with the examples, you will need access to a shell on your device.  This can be achieved with SSH.  Connect to your device with the command `ssh ovos@<device_ip_address>` and enter the password `ovos`.
```

Reviewer 2: ```suggestion
**NOTE** To continue with the examples, you will need access to a shell on your device. This can be achieved with SSH. Connect to your device with the command `ssh ovos@<device_ip_address>` and enter the password `ovos`.
```

That double space is erroneous too, right?

Author: I have always double spaced after a `.`  Thats how I was taught in highschool 30 years ago.  So ya, there are several of them

> This password is **EXTREAMLY** insecure and should be changed or use ssh keys for logging in.

Reviewer: ```suggestion
This password is **EXTREMELY** insecure and should be changed immediately. Better yet, use SSH keys for logging in.
```

Reviewer 2: Capitals *and* bold seems a little unnecessary.

> [Secureing SSH](config_security.md)

Reviewer: ```suggestion
[Securing SSH](config_security.md)
```

This section will explain how the configuration works, and how to do basic configuration changes.

The rest of this section will assume you have shell access to your device.

## How it works

OVOS will load configuration files from several locations and combine them into a single `json` file that is used throughout the software.

- `{ovos-config-path}/mycroft.conf`
  - usually in `<python_install_path>/site-packages/ovos_config/mycroft.conf`
  - This is the default configuration file distrubuted with `ovos-core`
- `os.environ.get('MYCROFT_SYSTEM_CONFIG')` or /etc/mycroft/mycroft.conf

>   - This is the default configuration file used by images, and may change specific values to corispond with how the image works.

Reviewer: ```suggestion
  - This is the default configuration file used by images, and may change specific values to correspond with how the image works.
```

- `os.environ.get('MYCROFT_WEB_CACHE')` or `XDG_CONFIG_PATH`/mycroft/web_cache.json
- `~/.mycroft/mycroft.conf` (Deprecated)
- `XDG_CONFIG_DIRS` + /mycroft/mycroft.conf
- `/etc/xdg/mycroft/mycroft.conf`
- `XDG_CONFIG_HOME` (default ~/.config) + /mycroft/mycroft.conf

>   - This is the file that you should use to modify the configuration.

Reviewer: Consider adding this note at the top of the list, or reversing the order of the list, to avoid confusion with users who read the first couple lines of documentation and end up editing the wrong config file.

Author: added a note to the top

When the configuration loader starts, it looks in these locations in this order, and loads ALL configurations. Keys that exist in multiple configuration files will be overridden by the last file to contain the value. This process results in a minimal amount being written for a specific device and user, without modifying default distribution files.

[Advanced Configuration Docs](https://openvoiceos.github.io/ovos-technical-manual/config/)

## Included Tools
OVOS provides a command line tool `ovos-config` for viewing and changing configuration values.

> Values can also be set manually in config files instead of using the CLI tool

Reviewer: ```suggestion
Values can also be set manually in config files instead of using the CLI tool.
```
